Automattic/wp-calypso
# Patterns: Show navigation bar header when user is logged-in

This pull request addresses https://github.com/Automattic/dotcom-forge/issues/5846 by making sure that the master bar showed at the top of the `Patterns` page has the right color scheme when the user is logged out (its background should have the same color than the header section that contains the title). It also adds the navigation bar header below that master bar when the user is logged in to be consistent with similar pages such as [Support](https://wordpress.com/support/):

##### Logged-out

Before | After
------ | -----
![image](https://github.com/Automattic/wp-calypso/assets/594356/e4392277-6bf6-4266-8568-b48688dd0d06) | ![image](https://github.com/Automattic/wp-calypso/assets/594356/c7c80806-57b9-47c1-9809-cb2109a6f758)

##### Logged-in
Before | After
------ | -----
![image](https://github.com/Automattic/wp-calypso/assets/594356/e9f7091f-0a3f-496e-9152-3093285379b2) | ![image](https://github.com/Automattic/wp-calypso/assets/594356/e401cdfa-42e8-4611-a34f-ba97221f0608)

Finally, it renames a few props types to be more consistent with the rest of our code.

#### Testing instructions

1. Run `git checkout update/patterns-header` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/88200#issuecomment-1979198273)
2. Open the [`Patterns` page](http://calypso.localhost:3000/patterns) while logged out
3. Assert that it looks like the screenshot above
4. Open that page again while logged in
5. Assert that it looks like the screenshot above

You may also want to check similar pages to make sure that no regression was introduced:

* http://calypso.localhost:3000/tags
* http://calypso.localhost:3000/themes/filter/business
* http://calypso.localhost:3000/site-profiler

You could also check how the `Patterns` page looks like on small viewports.
